### ISSUE TITLE
Hide all editor windows when focus is changed from the editor

### ISSUE THREAD

**Issue by itsaky** [AndroidIDEOfficial/AndroidIDE#490]
## Context
When changing between file tabs, the IDE manually notifies the editor to dismiss all popup windows that it has created. However, the editor should do this automatically when the focus is changed. This could be achieved by overriding the `onFocusChanged` method in [`IDEEdtor`](https://github.com/AndroidIDEOfficial/AndroidIDE/blob/main/app/src/main/java/com/itsaky/androidide/views/editor/IDEEditor.java) class.

## Expected behavior
The editor should dismiss it's windows automatically when the focus is changed.

## References
- [`onFocusChanged`](https://developer.android.com/reference/android/view/View#onFocusChanged(boolean,%20int,%20android.graphics.Rect)) function in the `View` class.
- [This is how the editor currently dismisses it's windows](https://github.com/AndroidIDEOfficial/AndroidIDE/blob/74b3c72e8a429202fa75190329ccf1749cddad51/app/src/main/java/com/itsaky/androidide/views/editor/IDEEditor.java#L669).

**Comment by Devendra34** [comment]
Hi, I would like to work on this issue, can you please assign it to me

**Comment by itsaky** [comment]
Sure.

**Comment by itsaky** [comment]
Any updates on this? Any issues?

**Comment by Devendra34** [comment]
> Any updates on this? Any issues?

Hi, there were gradle build issues on my system. 
I have resolved those, and I'll push a fix in some time.

**Comment by Devendra34** [comment]
Hi @itsaky,
Thanks for the detailed explanation for the issue. I think, I have resolved it in the same way how you were expecting. 
So depending upon the isFocus value from onFocusChanged() method, I'm calling the same ensureWindowsDismissed() method, which will dismiss all the windows same as before. Just the difference would be now I have deleted the code inside onTabUnselected() from EditorActivity.java. Hence we are not manually calling ensureWindowsDismissed() from the activity side.
Let me know if I'm missing something or if you want some more info on my implementation.
If the above things looks good to you, then I'll raise a PR on your confirmation.
Thanks!
